edb/schema/inheriting.py

```diff
@@ -770,17 +958,17 @@ def get_topmost_concrete_base(self, schema):
         raise errors.SchemaError(
             f'{self.get_verbosename(schema)} has no non-abstract ancestors')
 
-    def get_base_for_cast(self, schema):
-        if self.is_enum(schema):
+    def get_base_for_cast(self, schema: s_schema.Schema) -> so.Object:
+        if self.is_enum(schema):  # type: ignore
```

Comment:
Consider moving this check to the type that actually implements `is_enum`.

Reply:
Yes, it looks better :+1: